tests/test_scan_local_repo.py

```diff
@@ -32,3 +31,36 @@ def test_scan_exits_gracefully_when_target_is_not_git_repo(self):
                 str(result.exception),
                 f"Repository not found at {Path(run_path).resolve()}",
             )
+
+    def test_new_file_shows_up(self):
+        file_name = "secret_1.key"
+        runner = CliRunner()
+        # Add file with high entropy
+        secret_key = sha256(b"hello world")
+        with open(file_name, "a") as file:
+            file.write(secret_key.hexdigest())
+        repo = Repository(".")
+
+        # Check that tartufo picks up on newly added files
+        repo.index.add("tests/data/config/" + file_name)
+        repo.index.write() # This actually writes the index to disk. Without it, the tracked file is not actually staged.
+        result = runner.invoke(cli.main, ["--entropy-sensitivity", "1", "pre-commit"])
+        self.assertNotEqual(result.exit_code, 0)
```

Comment:
I finally realized what struck me as odd here. An exit code of `0` would indicate a successful scan with no issues found. So what I'm not sure about here is, how does this verify that the file was scanned? Since you're writing a sha256 digest to the file, that should get flagged for entropy, though I notice you modify the entropy sensitivity. Shouldn't we be letting it fail, and checking for an exit code greater than 0? Or even better maybe capture json output from the command and ensure that the filename in question is in the output?